lib/Conversion/TorchToLinalg/TorchToLinalg.cpp

```diff
@@ -2233,101 +2234,59 @@ class ConvertAtenUnsqueezeOp : public OpConversionPattern<AtenUnsqueezeOp> {
-        AffineMap::get(inputRank, 0, swapExprs, op.getContext())};
-    SmallVector<StringRef> iteratorTypes(inputRank, "parallel");
-    auto transpose = rewriter
-                         .create<linalg::GenericOp>(
-                             loc, outVector.getType(), inVector, outVector,
-                             indexingMaps, iteratorTypes,
-                             [](OpBuilder &b, Location loc, ValueRange args) {
-                               b.create<linalg::YieldOp>(loc, args[0]);
-                             })
-                         .getResult(0);
-    rewriter.replaceOpWithNewOp<tensor::CastOp>(op, outType, transpose);
-    return success();
-  }
-};
-} // namespace
-
-namespace {
-class ConvertAtenPermuteOp : public OpConversionPattern<AtenPermuteOp> {
+template <typename OpTy>
+class ConvertPermutationLikeOp : public OpConversionPattern<OpTy> {
 public:
-  using OpConversionPattern::OpConversionPattern;
+  using OpConversionPattern<OpTy>::OpConversionPattern;
   LogicalResult
-  matchAndRewrite(AtenPermuteOp op, ArrayRef<Value> operands,
+  matchAndRewrite(OpTy op, ArrayRef<Value> operands,
                   ConversionPatternRewriter &rewriter) const override {
     if (failed(verifyLinalgCompatibleTypes(op, rewriter)))
       return failure();
 
-    AtenPermuteOp::Adaptor adaptor(operands);
-    SmallVector<int64_t> dimensions;
-    if (!matchPattern(op.dims(), m_TorchConstantIntList(dimensions)))
-      return rewriter.notifyMatchFailure(op, "all dimensions must be constant");
+    Value inVector;
+    RankedTensorType inType;
+    int64_t inputRank = 0;
+    Type elementType;
 
-    Value inVector = adaptor.self();
-    auto inType = inVector.getType().cast<RankedTensorType>();
-    int64_t inputRank = inType.getRank();
-    auto outType = getTypeConverter()
+    auto fetchRankAndDtype = [&](Value inVector, RankedTensorType &inType,
+                                 int64_t &inputRank, Type &elementType) {
+      inType = inVector.getType().cast<RankedTensorType>();
+      inputRank = inType.getRank();
+      elementType = inType.getElementType();
+    };
+    SmallVector<int64_t> dimensions;
+    if (isa<AtenTransposeIntOp>(op)) {
+      auto transposeInt = cast<AtenTransposeIntOp>(op);
+      AtenTransposeIntOp::Adaptor adaptor(operands);
+      inVector = adaptor.self();
+      fetchRankAndDtype(inVector, inType, inputRank, elementType);
+      dimensions.resize(inputRank);
+      std::iota(dimensions.begin(), dimensions.end(), 0);
+      int64_t dim0Int, dim1Int;
+      if (!matchPattern(transposeInt.dim0(), m_TorchConstantInt(&dim0Int)) ||
+          !matchPattern(transposeInt.dim1(), m_TorchConstantInt(&dim1Int)))
+        return rewriter.notifyMatchFailure(
+            op, "The first two dimensions must be constant");
+      dimensions[toPositiveDim(dim0Int, inputRank)] = dim1Int;
+      dimensions[toPositiveDim(dim1Int, inputRank)] = dim0Int;
```

Comment:
Need to check `isValidDim`. If the user input an invalid dim, the program shouldn't crash.